feature-libs/checkout/core/events/checkout-event.listener.ts

```diff
@@ -42,6 +52,24 @@ export class CheckoutEventListener implements OnDestroy {
     );
   }
 
+  protected onSaveCartSuccess() {
+    this.subscription.add(
+      this.eventService.get(SaveCartSuccessEvent).subscribe((_event) => {
+        this.clearCheckoutFacade.resetCheckoutProcesses();
+      })
+    );
+  }
+
+  protected onRestoreSavedCartSuccess() {
```

Comment:
maybe these 2 functions can be merged into one just like `onUserAddressChange`. Listen to `SavedCartEvent`, and only if `SaveCartSuccessEvent` and `RestoreSavedCartSuccessEvent`, then resetCheckoutProcesses.

Reply:
I suggest we align the address book event to be consistent with the ones ini this PR.  Followup ticket GH-12630

Reply:
Fixed